### PR TITLE
Fix/delft3 dgt 874/875 login

### PR DESCRIPTION
**Pull request by CindyvdVries**

### CODE REVIEW

File: src/components/ConfirmDialog.vue [hunk 1]
```diff
@@ -11,13 +11,11 @@
...
           <slot name="body"></slot>
-          <div class="alert alert-info" role="alert" :id="`${dialogId}-dialog-alert`">
-            <slot name="alert"></slot>
-          </div>
+          <slot name="alert" class="alert alert-info" role="alert" :id="`${dialogId}-dialog-alert`"></slot>
         </div>
         <div class="modal-footer">
-          <button type="button" class="btn btn-warning" v-on:click.stop="confirm">{{ confirmButtonTitle }}</button>
-          <button type="button" class="btn btn-primary" v-on:click.stop="hide">Cancel</button>
+          <button type="button" class="btn btn-warning" @click="confirm">{{ confirmButtonTitle }}</button>
```
Comment (SiggyF) [inline src/components/ConfirmDialog.vue]: why was stop  there?

File: src/components/ImageAnimation.vue [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+          <li role="presentation"
+            class="nav-item"
+            :class="{'active': currentAnimationKey === 'delta_fringe_images'}"
+            @click="switchAnimation('delta_fringe_images')">
+            <a class="nav-link active" href="#">Delta fringe</a>
+          </li>
+          <li role="presentation"
+            class="nav-item"
+            :class="{'active': currentAnimationKey === 'channel_network_images'}"
+            @click="switchAnimation('channel_network_images')">
```
Comment (SiggyF) [inline src/components/ImageAnimation.vue]: switchTo? show? use a  "current" animation?

File: src/components/ModelCard.vue [hunk 3]
```diff
@@ -1,28 +1,29 @@
...
+  <div class="card-body" :class="{ 'card-active': model.active }" v-if="model.data">
+    <div class="row">
+      <div class="col-xs-8 col-md-6">
+        <span v-if="model.data.shared == 'p'"><i class="fa fa-fw fa-user" aria-hidden="true"></i></span>
+        <span v-if="model.data.shared == 'c'"><i class="fa fa-fw fa-group" aria-hidden="true"></i></span>
+        <span v-if="model.data.shared == 'w'"><i class="fa fa-fw fa-globe" aria-hidden="true"></i></span>
+        &nbsp;{{model.data.name}}
+      </div>
+      <div class="col-xs-4 col-md-3 my-auto">
+        <span class="badge" :class="[
```
Comment (SiggyF) [inline src/components/ModelCard.vue]: move  to computed?

File: src/components/ModelCard.vue [hunk 4]
```diff
@@ -101,32 +93,28 @@ export default {
...
-  div {
-    [class^='col-'] {
+    // Normally use mb-2 bootstrap css, but needed to remove this from last child
+    :not(:last-child.card-body) {
+      margin-bottom: 0.5rem;
+    }
+    :last-child.card-body {
       margin-bottom: 0;
-      padding-bottom: 0;
+      border-bottom: 1px solid #E2E8EB;
```
Comment (SiggyF) [inline src/components/ModelCard.vue]: colors from variables?

File: src/components/ScenarioCard.vue [hunk 5]
```diff
@@ -84,14 +71,11 @@ import store from '../store'
...
 import router from '../router.js'
-import {
-  bus
-} from '@/event-bus.js'
 
 export default {
   store,
   router,
-  template: '#template-scenario-card',
+  template: '#scenario-card',
```
Comment (SiggyF) [inline src/components/ScenarioCard.vue]: remove?

File: src/components/SearchList.vue [hunk 6]
```diff
@@ -58,71 +58,56 @@ export default {
...
-        // we only want the active ones
-        var activeItems = _.filter(allItems, ['active', true])
-
-        return activeItems
-      }
+    worldModels () {
+      // Get all models that are shared with the world
+      return _.filter(this.sharedState.modelContainers, ['data.shared', 'w'])
     }
+    // TODO: check why this code was here
```
Comment (SiggyF) [inline src/components/SearchList.vue]: remove?

File: src/components/SearchList.vue [hunk 7]
```diff
@@ -58,71 +58,56 @@ export default {
...
-    },
-    hasCompanyModels () {
-      return this.companyModels.length > 0
-    },
-    hasWorldModels () {
-      return this.worldModels.length > 0
-    },
-    action (thing) {
-      thing.active = !thing.active
+    notEmpty (arr) {
```
Comment (SiggyF) [inline src/components/SearchList.vue]: hmm...

File: src/components/UserDetails.vue [hunk 8]
```diff
@@ -27,5 +25,102 @@ export default {
 }
 </script>
 
-<style>
+<style lang="scss">
+@import '../assets/variables.scss';
+
+#dropdownMenuButton {
```
Comment (SiggyF) [inline src/components/UserDetails.vue]: avoid using id's in css.

File: src/store/index.js [hunk 9]
```diff
@@ -41,11 +35,77 @@ export default new Vuex.Store({
...
-    updateUser,
-    fetchUser,
+    fetchUser (context) {
+      if (context.state.reqUser !== undefined) {
+        context.state.reqUser.abort()
+      }
+      return new Promise((resolve, reject) => {
+        context.state.user = $.ajax({ url: 'api/v1/users/me/', data: context.state.params, traditional: true, dataType: 'json' })
+          .done(function (json) {
+            console.log('fetchuser worked', json)
```
Comment (SiggyF) [inline src/store/index.js]: cleanup console

File: src/store/index.js [hunk 10]
```diff
@@ -114,15 +174,14 @@ export default new Vuex.Store({
...
-
         if (container === undefined) {
           // create new container
           container = {
             id: model.id,
             active: false,
             selected: false,
             data: model,
-            statusLevel: this.statusLevel
+            statusLevel: model.state
```
Comment (SiggyF) [inline src/store/index.js]: rename  to state?

File: vue.config.js [hunk 11]
```diff
@@ -6,11 +6,11 @@ module.exports = {
   devServer: {
     port: 9000,
     proxy: {
-      '^/login': {
-        target: `${apiServer}/login/?next=/`
+      '^/login/*': {
```
Comment (SiggyF) [inline vue.config.js:9]: check regex syntax.

**Review by SiggyF — APPROVED** [review]

File: src/components/ImageAnimation.vue (hunk 2 above)
Comment (CindyvdVries) [reply src/components/ImageAnimation.vue]: first setup done, need to improve next sprint -> info needs to come from backend

**Pull request merged**